src/Widgets/MultitaskingView.vala

```diff
@@ -178,32 +178,34 @@ namespace Gala {
                 return true;
             }
 
-            if (scroll_event.direction != ScrollDirection.SMOOTH) {
+            if (scroll_event.direction == ScrollDirection.SMOOTH ||
```

Comment:
Was this right before?

This has changed from `!= SMOOTH` to `== SMOOTH`.

So if it is a smooth scroll event, we return here and let `GestureTracker` handle it, right? But does that mean this was wrong before? Or is there something I don't understand.

Reply:
The key here is that, no matter what kind of device produced the scroll, Mutter sends both a smooth and a non-smooth scroll events.

For example, 1 low-resolution ("normal" mouse) mouse wheel click will generate 1 scroll event with direction UP/DOWN *and* 1 scroll event with direction SMOOTH.

A touchpad or a high-resolution scroll mouse, scrolling a bit will generate X number of SMOOTH scroll events and, after a certain threshold is reached, 1 scroll event with direction != SMOOTH.

This design is very clever, because it allows to listen to discrete scroll events without worrying about the underlying hardware.

TL;DR

Answering your question, yes, before this change, this code worked. When the SMOOTH scroll events weren't handled by the `GestureTracker` (ie, non-touchpad events), this code was triggered and, by using a magic number, we changed the workspace.

However, once `xf86-input-libinput` 1.2.0 lands in elementary OS, we will receive SMOOTH scroll events from mice, and this won't work. Also, it is a good idea to avoid magic numbers and rely on Mutter's discrete scroll emulation.